### DISCUSSION TITLE
process.version undefined in middleware.ts

### DISCUSSION THREAD

**Discussion by leepowelldev** [vercel/next.js#51750] [Help]
### Summary

Wondering if this is a bug, or by design - but I cannot access certain `process` properties in my middleware function.

```Typescript
// /src/middleware.ts
console.log(process.version); // undefined
```

**Comment by danieljee** [comment]
I had the same problem. Middleware doesn't run in node.js runtime and node.js globals and some modules are not supported.

**Comment by leepowelldev** [comment]
I suspected it was something like that. I guess there's a reason for doing it - but rather frustrating.
